### PR TITLE
Create v0.39.0 RC branch.

### PR DESCRIPTION
**Pull request by multiphaseCFD**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Review by AmintorDusko — COMMENTED** [review]
So far, so good.

**Comment by multiphaseCFD** [comment]
Have to update device.rst for LT

File: pennylane_lightning/core/_version.py [hunk 1]
```diff
@@ -16,4 +16,4 @@
    Version number (major.minor.patch[-label])
 """
 
-__version__ = "0.39.0-dev51"
+__version__ = "0.39.0-rc0"
```
Comment (maliasadi) [inline pennylane_lightning/core/_version.py:19]:
```suggestion
__version__ = "0.39.0-rc1"
```

**Comment by maliasadi** [comment]
I'll check AARCH64 v0.39.0-rc1 wheels

**Comment by multiphaseCFD** [comment]
Confirmed `LGPU` and `LT` wheels on X86_64 work

File: pennylane_lightning/core/_version.py [hunk 2]
```diff
@@ -16,4 +16,4 @@
    Version number (major.minor.patch[-label])
 """
 
-__version__ = "0.39.0-dev51"
+__version__ = "0.39.0-rc1"
```
Comment (maliasadi) [inline pennylane_lightning/core/_version.py:19]:
```suggestion
__version__ = "0.39.0-rc2"
```

**Comment by maliasadi** [comment]
:partying_face:

**Pull request closed without merging**